spec/lib/msf/ui/console/command_dispatcher/core_spec.rb

```diff
@@ -397,4 +397,295 @@ def set_tabs_test(option)
       end
     end
   end
+
+  describe "#cmd_sessions" do
+    before(:each) do
+      allow(driver).to receive(:active_session=)
+      allow(framework).to receive(:sessions).and_return(sessions)
+      Timecop.freeze(Time.parse('Dec 18, 2022 12:33:40.000000000 GMT'))
+    end
+
+    context "with no sessions" do
+      let(:sessions) {[]}
+      it "should show an empty table with no args" do
+        core.cmd_sessions
+        expect(@combined_output.join("\n")).to match_table <<~TABLE
+          Active sessions
+          ===============
+          
+          No active sessions.
+        TABLE
+      end
+
+      it "should show no matches with a search term" do
+        core.cmd_sessions("--search", "session_id:1")
+        expect(@combined_output.join("\n")).to match_table <<~TABLE
+          No matching sessions.
+        TABLE
+      end
+    end
+
+    context "with sessions" do
+      let(:sessions) do 
+        {
+          1 => instance_double(::Msf::Sessions::Meterpreter_x64_Win, last_checkin: Time.now, type: 'meterpreter', sid:1, sname: "sesh1", info: "info", session_host: "127.0.0.1", tunnel_to_s: "tunnel")
+        }
+      end
+
+      it "returns an error searching for an invalid parameter" do
+        core.cmd_sessions("--search", "not_a_term:1")
+        expect(@combined_output.join("\n")).to match_table <<~TABLE
+          Please provide valid search term. Given: not_a_term
+        TABLE
+      end
+    end
+
+    context "searches with sessions with different ids" do
+
+      let(:sessions) do 
+        {
+          1 => instance_double(::Msf::Sessions::Meterpreter_x64_Win, last_checkin: Time.now, type: 'meterpreter', sid:1, sname: "session1", info: "info", session_host: "127.0.0.1", tunnel_to_s: "tunnel"),
+          2 => instance_double(::Msf::Sessions::Meterpreter_x64_Win, last_checkin: Time.now, type: 'meterpreter', sid:2, sname: "session2", info: "info", session_host: "127.0.0.1", tunnel_to_s: "tunnel"),
+        }
+      end
+
+      it "returns matching session when searching for id" do
+        core.cmd_sessions("--search", "session_id:1")
+        expect(@output.join("\n")).to match_table <<~TABLE
+          Active sessions
```

Comment:
This is a very minor gripe I noticed when reviewing this, having this say `Active Sessions` whether there has been a search or not might be a bit weird, that would imply to me that this is _all_ of the active sessions
Not a blocker, not sure what wording (if any) would be better

Reply:
This is a holdover from `dump_sessions` which is an existing function we're leveraging with Killall when used in conjunction with a search term. If it's not a blocker, I'll just try to see if there's a quick fix